### PR TITLE
Bartv/298 create user

### PR DESCRIPTION
**Pull request by bart-vandervliet**
Some changes we made to get create user working.

### CODE REVIEW

File: Core/App.Web/Builders/WebApplicationSetupBuilder.cs [hunk 1]
```diff
@@ -12,6 +12,8 @@ internal class WebApplicationSetupBuilder : ApplicationSetupBuilder, IWebApplica
     public IControllerOptions Controller { get => ControllerOptions; }
     internal ControllerOptions ControllerOptions { get; } = new ControllerOptions();
 
+    public ICollection<Type> RequestPipelineFilterMetadata { get; } = new List<Type>();
```
Comment (jstorteboom) [inline Core/App.Web/Builders/WebApplicationSetupBuilder.cs]: Maybe move this to the ControllerOptions and use a method AddFilters(params IFilterMetadata[] filters). This way multiple libs can start adding filters. To should then be gathered in an internal List<IFilterMetadata>. Then in StartupBase ln 82, you use that internal list to actually add those filters to the controller options. Would that make sense?

File: Core/Lib.Web/Builders/IWebApplicationSetupBuilder.cs [hunk 2]
```diff
@@ -8,4 +9,6 @@ public interface IWebApplicationSetupBuilder : IApplicationSetupBuilder
 {
     IControllerOptions Controller { get; }
     IHealthChecksBuilder HealthChecks { get; }
+    ICollection<Type> RequestPipelineFilterMetadata { get; }
```
Comment (jstorteboom) [inline Core/Lib.Web/Builders/IWebApplicationSetupBuilder.cs]: See previous comment on WebApplicationSetupBuilder.

**Review by jstorteboom — CHANGES_REQUESTED** [review]

File: Core/App.Web/Builders/WebApplicationSetupBuilder.cs (hunk 1 above)
Comment (Johanh-od) [reply Core/App.Web/Builders/WebApplicationSetupBuilder.cs]: Make sense, I'll update the code

Comment (Johanh-od) [reply Core/App.Web/Builders/WebApplicationSetupBuilder.cs]: Code updated

File: Core/App.Web/Options/ControllerOptions.cs [hunk 3]
```diff
@@ -30,4 +38,20 @@ public IControllerOptions UseViews()
         UsingViews = true;
         return this;
     }
+
+    public IControllerOptions AddRequestPipeLineFilter(Type filter)
```
Comment (jstorteboom) [inline Core/App.Web/Options/ControllerOptions.cs]: Mag dit ook zijn: 
public IControllerOptions AddRequestPipeLineFilter\<TFilter\>() where TFilter is IFilterMetadata.
of
public IControllerOptions AddRequestPipeLineFilter(params Type filter[])

Of als overloads beide opties? Ik denk dat het iets developer friendly is.

**Review by jstorteboom — APPROVED** [review]

**Pull request merged**